rest_framework_bulk/mixins.py

```diff
@@ -1,103 +1,12 @@
-                self.object = serializer.save(force_insert=True)
-                [self.post_save(obj, created=True) for obj in self.object]
-                return Response(serializer.data, status=status.HTTP_201_CREATED)
-
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
-
-
-class BulkUpdateModelMixin(object):
-    """
-    Update model instances in bulk by using the Serializer's
-    ``many=True`` ability from Django REST >= 2.2.5.
-    """
-
-    def get_object(self):
-        return self.get_queryset()
-
-    def bulk_update(self, request, *args, **kwargs):
-        partial = kwargs.pop('partial', False)
-
-        # restrict the update to the filtered queryset
-        serializer = self.get_serializer(self.filter_queryset(self.get_queryset()),
-                                         data=request.DATA,
-                                         many=True,
-                                         partial=partial)
-
-        if serializer.is_valid():
-            try:
-                [self.pre_save(obj) for obj in serializer.object]
-            except ValidationError as err:
-                # full_clean on model instances may be called in pre_save
-                # so we have to handle eventual errors.
-                return Response(err.message_dict, status=status.HTTP_400_BAD_REQUEST)
-            self.object = serializer.save(force_update=True)
-            [self.post_save(obj, created=False) for obj in self.object]
-            return Response(serializer.data, status=status.HTTP_200_OK)
-
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
-
-    def partial_bulk_update(self, request, *args, **kwargs):
-        kwargs['partial'] = True
-        return self.bulk_update(request, *args, **kwargs)
-
-
-class BulkDestroyModelMixin(object):
-    """
-    Destroy model instances.
-    """
-
-    def allow_bulk_destroy(self, qs, filtered):
-        """
-        Hook to ensure that the bulk destroy should be allowed.
-
-        By default this checks that the destroy is only applied to
-        filtered querysets.
-        """
-        return qs is not filtered
-
-    def bulk_destroy(self, request, *args, **kwargs):
-        qs = self.get_queryset()
-        filtered = self.filter_queryset(qs)
-        if not self.allow_bulk_destroy(qs, filtered):
-            return Response(status=status.HTTP_400_BAD_REQUEST)
-
-        for obj in filtered:
-            self.pre_delete(obj)
-            obj.delete()
-            self.post_delete(obj)
-        return Response(status=status.HTTP_204_NO_CONTENT)
+from __future__ import print_function, unicode_literals
+import rest_framework
+
+
+# import appropriate mixins depending on the DRF version
+# this allows to maintain clean code for each DRF version
+# without doing any magic
+# a little more code but a lit clearer what is going on
+if str(rest_framework.__version__).startswith('2'):
```

Comment:
`rest_framework.__version__.split('.')[0] == 2` might be slightly better as far as future-proofing.